axum/src/extract/path/de.rs

```diff
@@ -362,7 +375,18 @@ impl<'de> Deserializer<'de> for ValueDeserializer<'de> {
     where
         V: Visitor<'de>,
     {
-        visitor.visit_borrowed_str(self.value)
+        visitor
+            .visit_borrowed_str(self.value)
+            .map_err(|e: Self::Error| match e.kind {
+                ErrorKind::Message(message) => {
+                    PathDeserializationError::new(ErrorKind::DeserializeError {
+                        key: self.key.map(|k| k.key().to_owned()).unwrap_or_default(),
```

Comment:
When can this be `None`? Does using an empty string as a fallback make sense here? Can we maybe use another variant, even the catch all `Message` as a fallback instead?

Reply:
From what I understand of the code, it is only `None` when deserializing maps. This is also evident from this comment: https://github.com/tokio-rs/axum/blob/e3bb7083c886247f4e6931e149ef6067e6b82e1b/axum/src/extract/path/de.rs#L450-L451

I can match on the existence of the key, and keep it to `ErrorKind::Message` if key is ever `None`.